src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -1043,6 +1048,8 @@ private void HandleNodesCollectionChanges()
 
                 AddToCutGeometryDictionary(groupViewModel);
             }
+
+            AnnotationModel.UpdateErrorAndWarningIconVisibility();
```

Comment:
Whats the reason for calling this here? could we not just call it when nodes are added or removed from the AnnotationModel, here for example https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/Annotations/AnnotationModel.cs#L607